test/package.mjs

```diff
@@ -98,68 +97,6 @@ test('validate configuration', async t => {
 			`Configuration for "${name}" is invalid.`,
 		);
 	}
-
-	// `env`
-	{
-		// https://github.com/eslint/eslint/blob/32ac37a76b2e009a8f106229bc7732671d358189/conf/globals.js#L19
-		const testObjects = [
-			'undefinedGlobalObject',
-			// `es3`
-			'Array',
-			// `es5`
-			'JSON',
-			// `es2015`(`es6`)
-			'Promise',
-			// `es2021`
-			'WeakRef',
-		];
-		const baseOptions = {
-			useEslintrc: false,
-			plugins: {
-				unicorn: eslintPluginUnicorn,
-			},
-			overrideConfig: {
-				rules: {
-					'no-undef': 'error',
-				},
-			},
-		};
-		const getUndefinedGlobals = async options => {
-			const [{messages}] = await new ESLint({...baseOptions, ...options}).lintText(testObjects.join(';\n'));
-			return messages.map(({message}) => message.match(/^'(?<object>.*)' is not defined\.$/).groups.object);
-		};
-
-		t.deepEqual(await getUndefinedGlobals(), ['undefinedGlobalObject', 'Promise', 'WeakRef']);
-		t.deepEqual(await getUndefinedGlobals({baseConfig: eslintPluginUnicorn.configs.recommended}), ['undefinedGlobalObject']);
-
-		const availableEnvironments = [...eslintrc.Legacy.environments.keys()].filter(name => /^es\d+$/.test(name));
-		const recommendedEnvironments = Object.keys(eslintPluginUnicorn.configs.recommended.env);
-		t.is(recommendedEnvironments.length, 1);
-		t.is(
-			availableEnvironments.at(-1),
-			recommendedEnvironments[0],
-			'env should be the latest es version',
-		);
-	}
-
-	// `sourceType`
-	{
-		const text = 'import fs from "node:fs";';
-		const baseOptions = {
-			useEslintrc: false,
-			plugins: {
-				unicorn: eslintPluginUnicorn,
-			},
-		};
-		const runEslint = async options => {
-			const [{messages}] = await new ESLint({...baseOptions, ...options}).lintText(text);
-			return messages;
-		};
-
-		const [{message}] = await runEslint();
-		t.is(message, 'Parsing error: The keyword \'import\' is reserved');
-		t.deepEqual(await runEslint({baseConfig: eslintPluginUnicorn.configs.recommended}), []);
-	}
```

Comment:
`FlatESLint` uses the latest syntax by default now, I guess we should just remove these.